src/pages/store/StoreEdit.tsx

```diff
@@ -1,68 +1,88 @@
 import Button from '@/components/common/Button';
 import Modal from '@/components/common/Modal';
-import { getPresignedUrl, uploadImageToS3 } from '@/api/imageApi';
 import { AuthContext } from '@/context/AuthContext';
 
 type Category = (typeof CATEGORY_OPTIONS)[number];
 type Address1 = (typeof ADDRESS_OPTIONS)[number];
 
 interface StoreEditForm extends Omit<ShopRequest, 'category' | 'address1'> {
-  category: Category | '';
-  address1: Address1 | '';
+  category: Category | null;
+  address1: Address1 | null;
 }
 
-type InitialData = ShopItem | undefined;
-
 type ModalType = 'success' | 'warning' | 'auth';
 
 export default function StoreEdit() {
   const navigate = useNavigate();
-  const location = useLocation();
-  const initialData = location.state as InitialData;
   const { isLoggedIn } = useContext(AuthContext);
 
-  const isEditMode =
-    initialData !== undefined && typeof initialData?.id === 'string';
+  const shopId = localStorage.getItem('shopId');
+  const isEditMode = !!shopId;
 
-  const [edit, setEdit] = useState<StoreEditForm>(
-    initialData
-      ? {
-          name: initialData.name,
-          category: initialData.category,
-          address1: initialData.address1,
-          address2: initialData.address2,
-          description: initialData.description,
-          originalHourlyPay: initialData.originalHourlyPay,
-          imageUrl: initialData.imageUrl,
-        }
-      : {
-          name: '',
-          category: '',
-          address1: '',
-          address2: '',
-          description: '',
-          originalHourlyPay: 0,
-          imageUrl: '',
-        },
-  );
+  const [edit, setEdit] = useState<StoreEditForm>({
+    name: '',
+    category: null,
+    address1: null,
+    address2: '',
+    description: '',
+    originalHourlyPay: 0,
+    imageUrl: '',
+  });
 
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [modalContent, setModalContent] = useState('');
   const [modalType, setModalType] = useState<ModalType>('success');
 
+  // 로그아웃 처리 및 등록 수정 모드
+  useEffect(() => {
+    const userId = localStorage.getItem('userId');
+
+    if (!userId) {
+      setModalType('auth');
+      setModalContent('로그인이 필요합니다.');
+      setIsModalOpen(true);
+      return;
+    }
+
+    if (isEditMode && !shopId) {
+      setModalType('warning');
+      setModalContent('가게 정보를 찾을 수 없습니다.');
+      setIsModalOpen(true);
+      return;
+    }
+
+    if (!isEditMode) return;
+
+    const fetchShopInfo = async () => {
+      try {
+        const shopInfo = await getShop(shopId);
+        setEdit({
+          name: shopInfo.item.name ?? '',
+          category: shopInfo.item.category ?? null,
+          address1: shopInfo.item.address1 ?? null,
+          address2: shopInfo.item.address2 ?? '',
+          description: shopInfo.item.description ?? '',
+          originalHourlyPay: shopInfo.item.originalHourlyPay ?? 0,
+          imageUrl: shopInfo.item.imageUrl ?? '',
+        });
```

Comment:
```suggestion
        setEdit(shopInfo.item);
```
💬 이렇게 줄일 수 있을 것 같아요~

Reply:
오 좋네요! 바꾸겠습니다